### PR TITLE
EMTF Unpacker Update to Add HMT and Displaced Muon information

### PR DESCRIPTION
**Pull request by eyigitba**
#### PR description:

This PR updates the EMTF unpacker to add High Multiplicity Trigger (HMT) and Displaced Muon information. The EMTF firmware at P5 is already updated to add these parameters to EMTF DAQ output. 

The main changes are in `EMTFBlockSP` for unpacking the new information. The `EMTFCollections`, `EMTFSetup` `EMTFTokens` and `SP.h` are changed accordingly to include the new Regional Muon Shower object in the EMTF unpacker workflow. 

No changes are expected in performance. 

#### PR validation:

Validated with `runTheMatrix.py -l limited -i all --ibeos` and no changes are observed.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by epalencia** [comment]
Please test

**Comment by epalencia** [comment]
+l1

File: EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockSP.cc [hunk 1]
```diff
@@ -16,6 +16,13 @@ namespace l1t {
         bool unpack(const Block& block,
                     UnpackerCollections* coll) override;  // Apparently it's always good to use override in C++
         // virtual bool packBlock(const Block& block, UnpackerCollections *coll) override;
+
+      private:
+        bool useHMTBits_{false};
+        bool useNNBits_{false};
```
Comment (perrotta) [inline EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockSP.cc:22]:
```suggestion
```
No need to have the two bools here as class members: define them as local variables inside the `unpack` method, instead

File: EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockSP.cc [hunk 2]
```diff
@@ -16,6 +16,13 @@ namespace l1t {
...
+
+        int nominalShower_ = 2;
+        int tightShower_ = 3;
```
Comment (perrotta) [inline EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockSP.cc:25]:
```suggestion
```
Why `nominalShower_` and `tightShower_` are class members, if they are only used as constants in the `unpack` method?
Please make them `constexpr` and local inside `unpack`

File: EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockSP.cc [hunk 3]
```diff
@@ -176,6 +183,14 @@ namespace l1t {
         // payload[0] = bits 0-15, payload[1] = 16-31, payload[3] = 32-47, etc.
         auto payload = block.payload();
 
+        // FW version is computed as (Year - 2000)*2^9 + Month*2^5 + Day (see Block.cc and EMTFBlockTrailers.cc)
+        if (getAlgoVersion() >= 11098) {  // FW versions >= 26.10.2021
+          useNNBits_ = true;
+        }
+        if (getAlgoVersion() >= 11306) {  // FW versions >= 10.01.2022
+          useHMTBits_ = true;
+        }
```
Comment (perrotta) [inline EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockSP.cc:192]:
```suggestion
        bool  useNNBits_ =  getAlgoVersion() >= 11098;  // FW versions >= 26.10.2021
        bool useHMTBits_ = getAlgoVersion() >= 11306;  // FW versions >= 10.01.2022
        }
```

**Comment by eyigitba** [comment]
Thanks for the suggestions @perrotta! I implemented them.

**Comment by cecilecaillol** [comment]
please test

**Comment by epalencia** [comment]
+l1

**Comment by perrotta** [comment]
+1

**Pull request merged**